extensions/amp-consent/amp-consent.md

```diff
@@ -137,28 +151,125 @@ current user to the 3rd party vendor extensions.
+The consent decisions collected from user via this prompt UI will be stored in `localStorage` as client cache. See the [Client caching](#client-caching) section for how the cache is used.
 
-In the case that `checkConsentHref` and `promptIfUnknownForGeoGroup` are both defined. The value from `<amp-geo>` will be respected.
 
-#### promptUI
+#### geoOverride
+
+`geoOverride` provides a way to utilize the `<amp-geo>` component to detect user's geo location to assist client side decisions. 
+
+`geoOverride` is a JSON object keyed by geo group codes which are defined in `<amp-geo>` (details [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-geo/amp-geo.md)). Each geo override should be a valid `<amp-consent>` config object. AMP will take all the values in the corresponding `geoOverride` to override the existing config. 
+
+Two important tips when configuring `amp-geo`:
+
+- All geo groups should be mutually exclusive. The behavior is undetermined if a user falls into multiple geo override.
+- Provide an `geoGroupUnknown` override for users that are failed be be identified by `<amp-geo>`.
+
+Take the following config as an example:
+
+```json
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": false,
+
+  "geoOverride": {
+    "geoGroup1": {
+      "consentRequired": true,
+    }, 
+    "geoGroup2": {
+      "checkConsentHref": "https://example.com/check-consent",
+      "consentRequired": "remote",
+    },
+    "geoGroupUnknown": {
+      "checkConsentHref": "https://example.com/check-consent",
+      "consentRequired": true,
+   }
+  }
+}
+```
+
+For users outside `geoGroup1`, `geoGroup2` & `geoGroupUknown`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": "false",
+}
+```
+`<amp-consent>` does nothing because `"consentRequired": false`.
+
+
+For users in `geoGroup1`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": true,
+}
+```
+The previous consent state is read from local cache as `checkConsentHref` is not specified. Consent will be collected via the specified prompt UI if cache is empty.
+
+
+For users in `geoGroup2`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "checkConsentHref": "https://example.com/check-consent",
+  "consentRequired": "remote",
+}
+```
+If local cache is empty, AMP will wait for `checkConsentHref` response to decide whether a consent is required from the user. If the response contains `consentRequired: true` then AMP will collect consent via the specified prompt UI if consent state is `unknown`.
```

Comment:
nit: rewording

If the response contains `consentRequired: true` and `consentStateValue: unknown`, then AMP will collect consent via the specified prompt UI. If `consentStateValue` is 'accepted' or 'rejected', then it will use this value and also sync to cache.